online-shop/src/app/components/presentational/products-list-view/products-list-view.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Input } from '@angular/core';
+import { NavigationExtras, Router } from '@angular/router';
+import { Product } from 'src/app/types/products.types';
+import { ProductAndProductCategoryDto } from 'src/app/types/productsAndProductsCategoryDto';
+
+@Component({
+  selector: 'app-products-list-view',
+  templateUrl: './products-list-view.component.html',
+  styleUrls: ['./products-list-view.component.scss']
+})
+export class ProductsListViewComponent {
+  @Input()  productList :ProductAndProductCategoryDto[] | undefined;
+
+  constructor(private router: Router) {
+  }
+
+
+
+  showProductDetails(ProductAndProductCategoryDto: ProductAndProductCategoryDto): void {
+    const navigationExtras: NavigationExtras = {
```

Comment:
This isn't really a normal way to perform things in Angular. I'd rather you have the product ID within the query parameters of the details route (i.e., declare with :id when you declare the path), then take it from `this.route.snapshot.params.id`, then you get it from the mock thingy by ID. Usually, the list view contains just a subset of the properties of the object. Hence, commonly you will take the ID from the URL and perform a request to the backend to get the full details, rather than have it in the navigation stuff.